qiskit_nature/problems/sampling/protein_folding/qubit_op_builder.py

```diff
@@ -0,0 +1,544 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Builds qubit operators for all Hamiltonian terms in the protein folding problem."""
+from typing import Union
+
+import numpy as np
+from qiskit.opflow import OperatorBase, PauliOp, PauliSumOp
+
+from .bead_contacts.contact_map import ContactMap
+from .bead_distances.distance_map import DistanceMap
+from .exceptions.invalid_side_chain_exception import (
+    InvalidSideChainException,
+)
+from .exceptions.invalid_size_exception import InvalidSizeException
+from .penalty_parameters import PenaltyParameters
+from .peptide.pauli_ops_builder import _build_full_identity
+from .qubit_utils.qubit_fixing import _fix_qubits
+from .peptide.beads.base_bead import BaseBead
+from .peptide.peptide import Peptide
+
+
+# TODO link to a SamplingQubitOpBuilder interface
+def _build_qubit_op(
+    peptide: Peptide,
+    pair_energies: np.ndarray,
+    penalty_parameters: PenaltyParameters,
+    n_contacts: int,
+) -> Union[PauliSumOp, PauliOp]:
+    """
+    Builds a qubit operator for a total Hamiltonian for a protein folding problem. It includes
+    8 terms responsible for chirality, geometry and nearest neighbors interactions.
```

Comment:
Is this/should this be configurable?

Reply:
For now rather not. @pbark ?

Reply:
The model was defined based on all possible constrains we could think of. I do not think this needs to be configurable.

Reply:
I did not necessarily mean that it needs to be configurable to add more terms. I rather meant the scenario in which a user may wish to disable some of these terms.. Or is this nonsensical?